components/ui/Badge.tsx

```diff
@@ -13,16 +13,40 @@ export default function Badge({
   style,
   ...rest
 }: BadgeProps) {
-  const variantStyle = variant === "outline" ? styles.outline : styles.default
-  const customTextStyle =
-    variant === "outline" ? styles.outlineText : styles.defaultText
+  const variantStyle: Record<BadgeVariants, ViewStyle> = {
+    default: {
+      backgroundColor: theme.colors.black,
+      paddingVertical: 1,
+      paddingHorizontal: 6,
+    },
+    outline: {
+      paddingVertical: 4,
+      paddingHorizontal: 8,
+      borderColor: theme.colors.black,
+      borderWidth: 1,
+    },
+    primary: {
+      backgroundColor: theme.colors.white,
+      paddingVertical: 1,
+      paddingHorizontal: 8,
+    },
+  }
+
+  let customTextStyle
+  switch (variant) {
+    case "outline":
+      customTextStyle = styles.outlineText
+      break
+    case "primary":
+      customTextStyle = styles.primaryText
+      break
+    default:
+      customTextStyle = styles.defaultText
+  }
+
   return (
-    <View {...rest} style={[styles.container, variantStyle, style]}>
-      <CustomText
-        size="xs"
-        fontWeight={variant === "outline" ? "normal" : "bold"}
-        fontFamily="writer-mono"
-        style={[styles.text, customTextStyle]}>
+    <View style={[styles.container, variantStyle[variant], style]} {...rest}>
+      <CustomText size="xs" style={[styles.text, customTextStyle]}>
```

Comment:
what do you think of using an object for `customTextStyle` as we have for the `variantStyle`? something like this,
```
const customTextStyle= {
default: styles.defaultText
outline: styles.outlineText,
primary: styles.primaryText,
}
...
<View style={[styles.container, variantStyle[variant], style]} {...rest}>
      <CustomText size="xs" style={[styles.text, customTextStyle[variant]}>
      ...
```